src/linter/clippy.rs

```diff
@@ -40,6 +42,11 @@ impl Clippy {
         self
     }
 
+    pub fn set_preview(&mut self, preview: bool) -> &mut Self {
+        self.preview = preview;
+        self
+    }
+
     fn get_command_parameters(&self) -> Vec<&str> {
         let mut params = vec!["clippy", "--message-format", "json"];
```

Comment:
Is there any chance we could push 
```            
			"+nightly",
            "clippy-preview",
            "-Z",
            "unstable-options",
``` 
somewhere here if self.preview ? That would avoid us to have two separate functions (get_nightly_parameters and get_command_parameters)